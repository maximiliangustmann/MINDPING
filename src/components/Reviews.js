import React, { useState } from 'react'
import Filter from './Filter'
import Review from './Review'

export default function Reviews({ reviews, onRemove, onEdit }) {
  const filters = ['Show all', 'book', 'audiobook', 'podcast', 'film', 'series']
  const [active, setActive] = useState(filters[0])
  return (
    <>
      <Filter filters={filters} active={active} setActive={setActive} />
      {reviews
        ?.filter(
          (reviews) =>
            active === 'Show all' || reviews.category.includes(active)
        )
        .map((review) => (
          <Review
            key={review.id}
            {...review}
            onRemove={onRemove}
            onEdit={onEdit}
          />
        ))}
    </>
  )
}
