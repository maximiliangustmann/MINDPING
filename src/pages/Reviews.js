import React from 'react'
import Filter from '../components/Filter'
import Review from '../components/Review'
import Header from '../components/Header'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

Reviews.propTypes = {
  reviews: PropTypes.array,
  onRemove: PropTypes.func,
  onEdit: PropTypes.func,
  filters: PropTypes.array,
  active: PropTypes.string,
  setActive: PropTypes.func,
  activeFilter: PropTypes.func,
  sortBySubcategory: PropTypes.func,
  sortByRating: PropTypes.func,
}

export default function Reviews({
  reviews,
  onRemove,
  onEdit,
  filters,
  active,
  setActive,
  activeFilter,
  sortBySubcategory,
  sortByRating,
}) {
  return (
    <>
      <Header
        sortBySubcategory={sortBySubcategory}
        sortByRating={sortByRating}
      />
      <StyledReviews>
        {reviews
          ?.filter(
            (reviews) =>
              active === 'Show all' || reviews.category.includes(active)
          )
          .map((review) => (
            <Review
              key={review.id}
              {...review}
              onRemove={onRemove}
              onEdit={onEdit}
            />
          ))}
      </StyledReviews>

      <Filter
        filters={filters}
        active={active}
        setActive={setActive}
        activeFilter={activeFilter}
      />
    </>
  )
}

const StyledReviews = styled.div`
  display: grid;
  gap: 20px;
  @media (min-width: 750px) and (max-width: 1124px) {
    display: grid;
    grid-template-columns: 1fr 1fr;
  }
  @media (min-width: 1125px) {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
  }
`
