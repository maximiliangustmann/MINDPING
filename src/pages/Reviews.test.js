import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import Reviews from './Reviews'

describe('Reviews', () => {
  it('renders correctly', () => {
    const filters = [
      'Show all',
      'Book',
      'Audiobook',
      'Podcast',
      'Film',
      'Series',
    ]
    const tree = renderer.create(
      <MemoryRouter>
        <Reviews filters={filters} />
      </MemoryRouter>
    )
    expect(tree).toMatchSnapshot()
  })
})
