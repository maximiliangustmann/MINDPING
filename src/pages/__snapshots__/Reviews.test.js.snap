// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews renders correctly 1`] = `
Array [
  <div
    className="Header__HeaderWrapper-sc-1uw4t5o-2 bYSPhJ"
  >
    <a
      aria-current="page"
      className="active"
      data-cy="homebutton"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      <svg
        className="Header__HomeIcon-sc-1uw4t5o-0 jgwhff"
      >
        home.svg
      </svg>
    </a>
    <a
      aria-current={null}
      data-cy="addbutton"
      href="/create"
      onClick={[Function]}
    >
      <svg
        className="Header__AddIcon-sc-1uw4t5o-1 RFjBk"
        data-cy="add-icon"
      >
        add.svg
      </svg>
    </a>
    <div
      className="sc-AxirZ hlnYRa"
    >
      <svg
        className="sc-AxjAm fZvzJc"
        data-cy="sort-icon"
        onClick={[Function]}
      >
        sort.svg
      </svg>
    </div>
  </div>,
  <div
    className="Reviews__StyledReviews-sc-1ki3ws2-0 gHtrPU"
  />,
  <div
    className="Filter__FilterBarWrapper-sc-1lxxwpc-2 gzzGsf"
  >
    <div
      className="Filter__FilterBar-sc-1lxxwpc-0 NElOz"
      data-cy="filterbar"
    >
      <button
        className="Filter__FilterButton-sc-1lxxwpc-1 kXHlYl"
        onClick={[Function]}
      >
        Show all
      </button>
      <button
        className="Filter__FilterButton-sc-1lxxwpc-1 kXHlYl"
        onClick={[Function]}
      >
        Book
      </button>
      <button
        className="Filter__FilterButton-sc-1lxxwpc-1 kXHlYl"
        onClick={[Function]}
      >
        Audiobook
      </button>
      <button
        className="Filter__FilterButton-sc-1lxxwpc-1 kXHlYl"
        onClick={[Function]}
      >
        Podcast
      </button>
      <button
        className="Filter__FilterButton-sc-1lxxwpc-1 kXHlYl"
        onClick={[Function]}
      >
        Film
      </button>
      <button
        className="Filter__FilterButton-sc-1lxxwpc-1 kXHlYl"
        onClick={[Function]}
      >
        Series
      </button>
    </div>
  </div>,
]
`;
